forge/db/controllers/AccessToken.js

```diff
@@ -202,16 +206,21 @@ module.exports = {
      * Only the project and expiry date can be updated.
      * @param {Object} app - The app object
      * @param {Object} token - The token to update
-     * @param {Object|string} [project] - The project object or id (set to `null` or `undefined` to remove project scope)
+     * @param {'application'|'instance'} [autoAssignType] - The type of auto assign (set to `null` or `undefined` to remove auto assign)
+     * @param {string} [autoAssignId] - The auto assign id (set to `null` or `undefined` to remove auto assign)
      * @param {Date|'never'} [expiresAt] - The expiry date. If `undefined`, the token never expires
      */
-    updateTokenForTeamDeviceProvisioning: async function (app, token, project, expiresAt) {
+    updateTokenForTeamDeviceProvisioning: async function (app, token, autoAssignType, autoAssignId, expiresAt) {
         let scope = [...(token.scope || [])]
-        const projectId = (project && typeof project === 'object') ? project.id : project
-        // remove project scope & add updated project scope (if set)
+        const instanceId = autoAssignType === 'instance' ? (autoAssignId && typeof autoAssignId === 'object') ? autoAssignId.id : autoAssignId : null
+        const applicationId = autoAssignType === 'application' ? (autoAssignId && typeof autoAssignId === 'object') ? autoAssignId.hashid : autoAssignId : null
```

Comment:
Same as above

Reply:
Fair comment. I have altered this to be more verbose and clear in [code readability: remove double ternary logic](https://github.com/FlowFuse/flowfuse/pull/5394/commits/8d87abe8ba5bfdc9da4b31a68deecb497025f471)
[8d87abe](https://github.com/FlowFuse/flowfuse/pull/5394/commits/8d87abe8ba5bfdc9da4b31a68deecb497025f471)